Assets/Scripts/Gameplay/Managers/GlobalGameReferences.cs

```diff
@@ -82,29 +92,78 @@ private void Update()
             }
         }
 
-        private void CheckReferences()
+        private void CheckAndSetReferences()
         {
             if (_ballController == null)
             {
                 LogNullRef(typeof(BallController).Name, LogType.Error);
+                _ballController = FindObjectOfType<BallController>();
+            }
+
+            if (_ballCockpit == null)
+            {
+                LogNullRef(typeof(BallCockpit).Name, LogType.Error);
+                _ballCockpit = FindObjectOfType<BallCockpit>();
+            }
+
+            if (_checkpoints == null)
+            {
+                LogNullRef("Checkpoints", LogType.Error);
+            }
+
+            if (_profileManager == null)
+            {
+                LogNullRef(typeof(ProfileManager).Name, LogType.Error);
+                _profileManager = FindObjectOfType<ProfileManager>();
+            }
+
+            if (_uiManager == null)
+            {
+                LogNullRef(typeof(UIManager).Name, LogType.Error);
+                _uiManager = FindObjectOfType<UIManager>();
+            }
+
+            if (_playerRigidbody == null)
+            {
+                LogNullRef("Player Rigidbody", LogType.Error);
             }
 
             if (_cameraController == null)
             {
                 LogNullRef(typeof(CameraController).Name, LogType.Error);
+                _cameraController = FindObjectOfType<CameraController>();
+            }
+
+            if (_powerupManager == null)
+            {
+                LogNullRef(typeof(PowerupManager).Name, LogType.Error);
+                _powerupManager = FindObjectOfType<PowerupManager>();
             }
 
             if (_gameSFXManager == null)
             {
+                LogNullRef(typeof(GameSFXManager).Name, LogType.Warning);
                 _gameSFXManager = FindObjectOfType<GameSFXManager>();
+            }
+
+            if (_levelManager == null)
+            {
+                LogNullRef(typeof(LevelManager).Name, LogType.Error);
+                _levelManager = FindObjectOfType<LevelManager>();
             }
+
         }
 
         private void PopulateArrays()
         {
             _checkpoints = GameObject.FindGameObjectsWithTag(Constants.CHECKPOINT_TAG);
         }
 
+        public void SetIsFollowingSpline(bool setting)

```

Comment:
I would change the name 'setting' to something like 'value', might be a bit more readable as value, but what do you think?

Reply:
Sure, changed to that.